src/tasks/task-se3-equality.cpp

```diff
@@ -199,8 +206,18 @@ namespace tsid
       m_v_ref_vec = m_v_ref.toVector();
       m_v = v_frame.toVector();
 
-      m_constraint.setMatrix(m_mask.asDiagonal() * m_J);
-      m_constraint.setVector(m_mask.asDiagonal() * (m_a_des - m_drift.toVector()));
+
+      m_a_des -= m_drift.toVector();
+
+      int idx = 0;
+      for (int i = 0; i < 6; i++) {
+        if (m_mask(i) != 1.) continue;
+
+        m_constraint.matrix().row(idx) = m_J.row(idx);
+        m_constraint.vector().row(idx) = m_a_des.row(idx);
```

Comment:
I think you used the wrong index when accessing `m_J` and `m_a_des`. It should be:
```
m_constraint.matrix().row(idx) = m_J.row(i);
m_constraint.vector().row(idx) = m_a_des.row(i);
```
Probably you couldn't see it in your tests because your mask has all the 1's first, and all the 0's last, and in this case the two implementations are equivalent. If you add a test where the mask is [0, 0, 0, 1, 1, 1]  you should see the difference.